openpype/hosts/houdini/plugins/publish/extract_active_view_thumbnail.py

```diff
@@ -0,0 +1,55 @@
+import pyblish.api
+
+from openpype.pipeline import publish, registered_host
+from openpype.hosts.houdini.api import lib
+from openpype.hosts.houdini.api.pipeline import IS_HEADLESS
+
+
+class ExtractActiveViewThumbnail(publish.Extractor):
+    """Set instance thumbnail to a screengrab of current active viewport.
+
+    This makes it so that if an instance does not have a thumbnail set yet that
+    it will get a thumbnail of the currently active view at the time of
+    publishing as a fallback.
+
+    """
+    order = pyblish.api.ExtractorOrder + 0.49
+    label = "Extract Active View Thumbnail"
+    families = ["workfile"]
+    hosts = ["houdini"]
+
+    def process(self, instance):
+        if IS_HEADLESS:
+            self.log.debug(
+                "Skip extraction of active view thumbnail, due to being in"
+                "headless mode."
+            )
+            return
+
+        thumbnail = instance.data.get("thumbnailPath")
+        self.log.debug(thumbnail)
```

Comment:
```suggestion
        self.log.debug("Extracting thumbnail path: {}".format(thumbnail))

Maybe it's better to have debug message to explain what `thumbnail` represents?

Reply:
I actually think this debug log was there only for development.
```suggestion
```

Reply:
Yup, I'd also remove this debug statement.